test/suites/app-service-configurable/main_test.go

```diff
@@ -0,0 +1,51 @@
+package test
+
+import (
+	"edgex-snap-testing/test/utils"
+	"log"
+	"os"
+	"testing"
+)
+
+func TestMain(m *testing.M) {
+
+	log.Println("[SETUP]")
+
+	// start clean
+	utils.SnapRemove(nil,
+		"edgex-app-service-configurable",
+		"edgexfoundry",
+	)
+
+	// install the app-service-configurable snap before edgexfoundry
+	// to catch build error sooner and stop
+	if utils.LocalSnap != "" {
+		utils.SnapInstallFromFile(nil, utils.LocalSnap)
+	} else {
+		utils.SnapInstallFromStore(nil, "edgex-app-service-configurable", utils.ServiceChannel)
+	}
+	utils.SnapInstallFromStore(nil, "edgexfoundry", utils.PlatformChannel)
+
+	// for local build, the interface isn't auto-connected.
+	// connect manually regardless
+	utils.SnapConnect(nil,
+		"edgexfoundry:edgex-secretstore-token",
+		"edgex-app-service-configurable:edgex-secretstore-token",
+	)
+
+	// set profile to rules engine
+	utils.Exec(nil, "sudo snap set edgex-app-service-configurable profile=rules-engine")
```

Comment:
This is something that we should have tests for.

Does ASC default to some profile when this isn't set? What happens when profile is set to "default"?

See https://github.com/edgexfoundry/app-service-configurable/blob/0eb39531cc0e251679d567441103b89f8fe292d6/snap/local/runtime-helpers/bin/service-wrapper.sh

The profile value is passed to the service command. We may be able to verify that a profile is set by looking at the ASC systemctl service file. Need to check.

Reply:
Yes, we could query the snap service logs and look for a record of the expected profile.

Reply:
> What do you mean by "snap opinions processing"?

I was referring to https://github.com/edgexfoundry/app-service-configurable/blob/0eb39531cc0e251679d567441103b89f8fe292d6/snap/local/runtime-helpers/bin/service-wrapper.sh#L5-L12

This is a wrapper script and its last line starts the application. In contrast, in many other services we process the options in snapcraft 's command-chain and pass them to the "command". In the latter case, it is possible to inspect the systemctl files and find out what is the service command that is being executed. For ASC, we can't do that right now.


> 
> ASC does not default to any profile when this isn't set:
> 
> ```
> level=INFO ts=2022-03-31T08:48:37.962447293Z app=app-<profile> source=service.go:485 msg="Starting app- 2.2.0-dev.23 "
> level=ERROR ts=2022-03-31T08:48:37.962517969Z app=app- source=bootstrap.go:46 msg="could not load service configuration file (/var/snap/edgex-app-service-configurable/1125/config/res/configuration.toml): open /var/snap/edgex-app-service-configurable/1125/config/res/configuration.toml: no such file or directory"
> ```
> 
> When the profile is set to "default", there will be no error logs anymore, but it is not possible to make the service running either:
> 
> ```
> $ sudo snap set edgex-app-service-configurable profile=default
> systemd[1]: snap.edgex-app-service-configurable.app-service-configurable.service: Failed with result 'exit-code'.
> systemd[1]: Failed to start Service for snap application edgex-app-service-configurable.app-service-configurable.
> ```

Got it. Thanks for checking. Looks like it should now be "simple" instead of default. See https://docs.edgexfoundry.org/2.1/microservices/application/AppServiceConfigurable/#deploying-multiple-instances-using-profiles

Let's leave the rules engine in main test for now until we fix that on ASC's snap side.

Reply:
> Yes, we could query the snap service logs and look for a record of the expected profile.

Ok. Please add (back) the SnapLogs function which returned stdout as string, and then do a string search for the profile name. The new config test function would set a profile and check that it was taken by the service. What do you think?

Reply:
Thanks. Okay for now. 

Note that the profile isn't a config env subtest. It should be a standalone test called e.g. `TestProfileConfig`>

Reply:
For the sleep, you can use time.Sleep(1*time.Second).

Reply:
`time := time.Now()` -> `start = time.Now()` so you don't use the time package name for a variable.